### PR TITLE
Issue #9580: update example of AST for TokenTypes.SWITCH_RULE

### PR DESCRIPTION
**Pull request by jz20**
Closes: #9580

Generated javadoc:
javadoc-1
javadoc-2

Code sample:
```
$ javac ~/Test.java
$ cat ~/Test.java
public class Test {

  public static String test(Day day) {
    return switch (day) {
      case SAT, SUN -> "Weekend";
      default -> "Working day";
    };
  }
  
  enum Day {
    SAT,
    SUN,
    OTHER
  }

}
```

Relevant AST:
```
    |       |--LITERAL_RETURN -> return [4:4]
    |       |   |--EXPR -> EXPR [4:11]
    |       |   |   `--LITERAL_SWITCH -> switch [4:11]
    |       |   |       |--LPAREN -> ( [4:18]
    |       |   |       |--EXPR -> EXPR [4:19]
    |       |   |       |   `--IDENT -> day [4:19]
    |       |   |       |--RPAREN -> ) [4:22]
    |       |   |       |--LCURLY -> { [4:24]
    |       |   |       |--SWITCH_RULE -> SWITCH_RULE [5:6]
    |       |   |       |   |--LITERAL_CASE -> case [5:6]
    |       |   |       |   |   |--EXPR -> EXPR [5:11]
    |       |   |       |   |   |   `--IDENT -> SAT [5:11]
    |       |   |       |   |   |--COMMA -> , [5:14]
    |       |   |       |   |   `--EXPR -> EXPR [5:16]
    |       |   |       |   |       `--IDENT -> SUN [5:16]
    |       |   |       |   |--LAMBDA -> -> [5:20]
    |       |   |       |   |--EXPR -> EXPR [5:23]
    |       |   |       |   |   `--STRING_LITERAL -> "Weekend" [5:23]
    |       |   |       |   `--SEMI -> ; [5:32]
    |       |   |       |--SWITCH_RULE -> SWITCH_RULE [6:6]
    |       |   |       |   |--LITERAL_DEFAULT -> default [6:6]
    |       |   |       |   |--LAMBDA -> -> [6:14]
    |       |   |       |   |--EXPR -> EXPR [6:17]
    |       |   |       |   |   `--STRING_LITERAL -> "Working day" [6:17]
    |       |   |       |   `--SEMI -> ; [6:30]
    |       |   |       `--RCURLY -> } [7:4]
    |       |   `--SEMI -> ; [7:5]
```

Complete AST:
```
$ java -jar target/checkstyle-8.42-SNAPSHOT-all.jar -t ~/Test.java
CLASS_DEF -> CLASS_DEF [1:0]
|--MODIFIERS -> MODIFIERS [1:0]
|   `--LITERAL_PUBLIC -> public [1:0]
|--LITERAL_CLASS -> class [1:7]
|--IDENT -> Test [1:13]
`--OBJBLOCK -> OBJBLOCK [1:18]
    |--LCURLY -> { [1:18]
    |--METHOD_DEF -> METHOD_DEF [3:2]
    |   |--MODIFIERS -> MODIFIERS [3:2]
    |   |   |--LITERAL_PUBLIC -> public [3:2]
    |   |   `--LITERAL_STATIC -> static [3:9]
    |   |--TYPE -> TYPE [3:16]
    |   |   `--IDENT -> String [3:16]
    |   |--IDENT -> test [3:23]
    |   |--LPAREN -> ( [3:27]
    |   |--PARAMETERS -> PARAMETERS [3:28]
    |   |   `--PARAMETER_DEF -> PARAMETER_DEF [3:28]
    |   |       |--MODIFIERS -> MODIFIERS [3:28]
    |   |       |--TYPE -> TYPE [3:28]
    |   |       |   `--IDENT -> Day [3:28]
    |   |       `--IDENT -> day [3:32]
    |   |--RPAREN -> ) [3:35]
    |   `--SLIST -> { [3:37]
    |       |--LITERAL_RETURN -> return [4:4]
    |       |   |--EXPR -> EXPR [4:11]
    |       |   |   `--LITERAL_SWITCH -> switch [4:11]
    |       |   |       |--LPAREN -> ( [4:18]
    |       |   |       |--EXPR -> EXPR [4:19]
    |       |   |       |   `--IDENT -> day [4:19]
    |       |   |       |--RPAREN -> ) [4:22]
    |       |   |       |--LCURLY -> { [4:24]
    |       |   |       |--SWITCH_RULE -> SWITCH_RULE [5:6]
    |       |   |       |   |--LITERAL_CASE -> case [5:6]
    |       |   |       |   |   |--EXPR -> EXPR [5:11]
    |       |   |       |   |   |   `--IDENT -> SAT [5:11]
    |       |   |       |   |   |--COMMA -> , [5:14]
    |       |   |       |   |   `--EXPR -> EXPR [5:16]
    |       |   |       |   |       `--IDENT -> SUN [5:16]
    |       |   |       |   |--LAMBDA -> -> [5:20]
    |       |   |       |   |--EXPR -> EXPR [5:23]
    |       |   |       |   |   `--STRING_LITERAL -> "Weekend" [5:23]
    |       |   |       |   `--SEMI -> ; [5:32]
    |       |   |       |--SWITCH_RULE -> SWITCH_RULE [6:6]
    |       |   |       |   |--LITERAL_DEFAULT -> default [6:6]
    |       |   |       |   |--LAMBDA -> -> [6:14]
    |       |   |       |   |--EXPR -> EXPR [6:17]
    |       |   |       |   |   `--STRING_LITERAL -> "Working day" [6:17]
    |       |   |       |   `--SEMI -> ; [6:30]
    |       |   |       `--RCURLY -> } [7:4]
    |       |   `--SEMI -> ; [7:5]
    |       `--RCURLY -> } [8:2]
    |--ENUM_DEF -> ENUM_DEF [10:2]
    |   |--MODIFIERS -> MODIFIERS [10:2]
    |   |--ENUM -> enum [10:2]
    |   |--IDENT -> Day [10:7]
    |   `--OBJBLOCK -> OBJBLOCK [10:11]
    |       |--LCURLY -> { [10:11]
    |       |--ENUM_CONSTANT_DEF -> ENUM_CONSTANT_DEF [11:4]
    |       |   |--ANNOTATIONS -> ANNOTATIONS [11:4]
    |       |   `--IDENT -> SAT [11:4]
    |       |--COMMA -> , [11:7]
    |       |--ENUM_CONSTANT_DEF -> ENUM_CONSTANT_DEF [12:4]
    |       |   |--ANNOTATIONS -> ANNOTATIONS [12:4]
    |       |   `--IDENT -> SUN [12:4]
    |       |--COMMA -> , [12:7]
    |       |--ENUM_CONSTANT_DEF -> ENUM_CONSTANT_DEF [13:4]
    |       |   |--ANNOTATIONS -> ANNOTATIONS [13:4]
    |       |   `--IDENT -> OTHER [13:4]
    |       `--RCURLY -> } [14:2]
    `--RCURLY -> } [16:0]
```

Expected javadoc update:
```
LITERAL_RETURN -> return
 |--EXPR -> EXPR
 |   `--LITERAL_SWITCH -> switch
 |       |--LPAREN -> (
 |       |--EXPR -> EXPR
 |       |   `--IDENT -> day
 |       |--RPAREN -> )
 |       |--LCURLY -> {
 |       |--SWITCH_RULE -> SWITCH_RULE
 |       |   |--LITERAL_CASE -> case
 |       |   |   |--EXPR -> EXPR
 |       |   |   |   `--IDENT -> SAT
 |       |   |   |--COMMA -> ,
 |       |   |   `--EXPR -> EXPR
 |       |   |       `--IDENT -> SUN
 |       |   |--LAMBDA -> ->
 |       |   |--EXPR -> EXPR
 |       |   |   `--STRING_LITERAL -> "Weekend"
 |       |   `--SEMI -> ;
 |       |--SWITCH_RULE -> SWITCH_RULE
 |       |   |--LITERAL_DEFAULT -> default
 |       |   |--LAMBDA -> ->
 |       |   |--EXPR -> EXPR
 |       |   |   `--STRING_LITERAL -> "Working day"
 |       |   `--SEMI -> ;
 |       `--RCURLY -> }
 `--SEMI -> ;
```

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java [hunk 1]
```diff
@@ -4359,39 +4359,39 @@
...
-     *   |       |   `--SEMI (;)
-     *   |       |--SWITCH_RULE
-     *   |       |   |--LITERAL_DEFAULT (default)
-     *   |       |   |--LAMBDA {@code ->}
-     *   |       |   |--EXPR
-     *   |       |   |   `--STRING_LITERAL ("Working day")
-     *   |       |   `--SEMI (;)
-     *   |       `--RCURLY (})
-     *   `--SEMI (;)
+     *  LITERAL_RETURN -&gt; return
```
Comment (timurt) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:4368]: lets put `--` in front of `LITERAL_RETURN`, and shift right following lines

**Review by timurt — CHANGES_REQUESTED** [review]

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java (hunk 1 above)
Comment (jz20) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java]: Thanks for the review. I have made the requested change.

**Review by timurt — APPROVED** [review]

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java [hunk 2]
```diff
@@ -4359,39 +4359,39 @@
...
-     *   |       |   `--SEMI (;)
-     *   |       |--SWITCH_RULE
-     *   |       |   |--LITERAL_DEFAULT (default)
-     *   |       |   |--LAMBDA {@code ->}
-     *   |       |   |--EXPR
-     *   |       |   |   `--STRING_LITERAL ("Working day")
-     *   |       |   `--SEMI (;)
-     *   |       `--RCURLY (})
-     *   `--SEMI (;)
+     *  --LITERAL_RETURN -&gt; return
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:4368]: Remove leading '--' to indent whole bloy to left

**Review by romani — CHANGES_REQUESTED** [review]
Item

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java (hunk 2 above)
Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:4368]: Sorry that we are conflicting in request, but we do not need such symbols

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java (hunk 1 above)
Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java]: @timurt , as we agreed in other PR, we do not need such leading indentation.

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java (hunk 2 above)
Comment (jz20) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java]: No worries. I have made the requested changes.

**Review by romani — APPROVED** [review]

**Pull request merged**